metabase/metabase
# Add tooltip displaying the exact time of the last collection item edit

Adds a tooltip displaying the exact time of the most recent edit, as the table only displays day, month, and year.

Also, this PR adds a `DateTime` component that formats date-time respecting the user's Metabase localization settings (stuff like 12h-clock vs. 24h-clock, date formats, etc.). I couldn't find an appropriate helper for that. ⚠️ There is a high chance I reinvented the wheel here. Please LMK if there is a helper for that already. If not, maybe it's worth checking the l10n settings are respected in different areas of the application.

A bit more on a11y:
* it's now possible to set an a11y role for a `Popover` component
* all tooltips will now have `role="tooltip"`

### To Verify

~⚠️ You may notice that the date and time do not update once you actually edit something. It's a known BE issue, and there is a PR for that already #16520~ (merged and rebased, should work as expected now)

1. Open a collection page
2. Hover any item's "last edited at" cell
3. A tooltip displaying a timestamp like "June 3, 2021, 10:12 AM" should appear

### Demo

<img width="1012" alt="Screenshot 2021-06-08 at 18 08 32" src="https://user-images.githubusercontent.com/17258145/121210860-b2723700-c884-11eb-85e7-1f7193206d49.png">
